giraffe/src/utils/PlotEnv.ts

```diff
@@ -149,22 +150,13 @@ export class PlotEnv {
     )
   }
 
-  public get yColumnType(): ColumnType {
-    //added to allow the y-axis type to be a string
-    //if there are multiple layers (ex. single stat + line graph),
-    //it will pick the yColumnType of the 1st layer (so BEWARE)
-    for (let i = 0; i < this.config.layers.length; i++) {
-      const layer: any = this.config.layers[i]
-      if (layer.yColumnType) {
-        return layer.yColumnType
-      }
-    }
+  public get yColumnLabels(): string[] {
+    const currLayerSpec = this.config.layers.map((_, i) => this.getSpec(i))[0]
 
-    return 'number'
-  }
-
-  public set yColumnType(columnType: ColumnType) {
-    this.config.yColumnType = columnType
+    if (currLayerSpec.type === 'mosaic') {
```

Comment:
ahh, okay, it makes sense why this doesn't bother TypeScript. :+1:

Not relevant to shipping this code but from a software readability / design perspective, I'm less of a fan of this. It's ultimately fine, because TypeScript is guaranteeing this things safety, but it makes it less explicit and more work for the reader's comprehension.

This pattern forces the reader to look up the definition for mosaic to see that `yColumnLabels` will always be an array on that, vs having the code here explicitly say "if Array.isArray(currLayerSpec.yColumnLabels), return that array, otherwise return an empty array."

Ultimately minor thing in code review, but has a larger cost down the line for anyone reading this. Like I said, not something to worry about in this review, but something to think about from a larger picture perspective.